documentation/API_documentation/webrtc voice api.md

```diff
@@ -0,0 +1,102 @@
+# Overview
+
+Bring your own number (BYON) service aims to provide consumers with the ability to use their phone number (MSISDN) for outgoing and incoming calls from non-SIM card devices. To access the service, users are required to authenticate using their digital credentials associated with their MNO account, in example their operator app credentials. As registration from a web endpoint is necessary to enable the service on non-SIM card devices.
+
+Outgoing calls made through the service will display the user's MSISDN as the caller number, ensuring that the recipient recognizes the user's phone number on their caller ID.
+
+Incoming calls introduce a unique functionality whereby multiple devices will ring simultaneously. This includes the user's smartphone and the web endpoint they have registered from. This design enables users to conveniently answer calls from any of these devices, depending on their preference or accessibility. The service is compatible with various devices that support web browsing environments, such as laptops, PCs, and XR immersive glasses.
+
+The service specification emphasises the accessibility and flexibility of utilising one's phone number across different non-SIM card devices. By leveraging this web-based service, users can extend their communication capabilities beyond traditional SIM-based devices, enhancing their overall communication experience.
+
+## Introduction
+
+This service brings new functionalities for the WebRTC services whose idea is to integrate BYON API to maintain the client number in out-bound and in-bound calls from web comms service.
+
+**Out-bound calls:**
+
+- Recipient receives call as if it was from User's normal phone number.
+- Maintains common number through web comms service and standard subscription.
+- Facilitates number recognition and call-back for co-workers, partners and/or clients.
+
+**In-bound calls:**
+
+- Receive phone calls simultaneously on phone subscription and web comms service.
+- Presents 'Occupied user' info to caller user is already in phone call.
+- Facilitates common multi-channel contact with co-workers, partners and/or clients.
+
+## API Spec
+
+### Roles, Actors and Scope
+
+Roles: Customer-user
+
+Actors: Application service providers, hyperscalers, application developers.
+
+Scope:
+
+1. Register to Network – Register the client endpoint instance to MNO's IMS Network.
+2. Create Notification Channel – Create Notification channel of specified type to receive notifications from the network (e.g. incoming call).
```

Comment:
We might have to include independent notification channel provider.

Reply:
The API is independent of whether it is implemented by independent channel provider or WebRTC gateway.  the API definition will remain the same, rest is implementation dependent.